cli/src/config.rs

```diff
@@ -0,0 +1,104 @@
+use clap::Parser;
+use color_eyre::Result;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use color_eyre::eyre::bail;
+
+#[derive(Parser, Debug)]
+#[allow(clippy::struct_excessive_bools)]
+pub struct Args {
+    /// Defines where your TS bindings will be saved by setting TS_RS_EXPORT_DIR
+    #[arg(long, short)]
+    pub output_directory: PathBuf,
+
+    /// Disables warnings caused by using serde attributes that ts-rs cannot process
+    #[arg(long)]
+    pub no_warnings: bool,
+
+    /// Adds the ".js" extension to import paths
+    #[arg(long)]
+    pub esm_imports: bool,
+
+    /// Formats the generated TypeScript files
+    #[arg(long)]
+    pub format: bool,
+
+    /// Generates an index.ts file in your --output-directory that re-exports all
+    /// types generated by ts-rs
+    #[arg(long = "index")]
+    pub generate_index_ts: bool,
+
+    /// Generates only a single index.ts file in your --output-directory that
+    /// contains all exported types
+    #[arg(long = "merge")]
+    pub merge_files: bool,
+
+    /// Do not capture `cargo test`'s output, and pass --nocapture to the test binary
+    #[arg(long = "nocapture")]
+    pub no_capture: bool,
+}
+
+// keeping this separate from `Args` for now :shrug:
+#[derive(Default, Deserialize)]
+#[serde(deny_unknown_fields, default)]
+pub struct Config {
+    // type overrides for types implemented inside ts-rs.
+    pub overrides: HashMap<String, String>,
+    pub output_directory: PathBuf,
+    pub no_warnings: bool,
+    pub esm_imports: bool,
+    pub format: bool,
+    pub generate_index_ts: bool,
+    pub merge_files: bool,
+    pub no_capture: bool,
+}
+
+impl Config {
+    pub fn load() -> Result<Self> {
+        let mut cfg = Self::load_from_file()?;
+        cfg.merge(Args::parse());
+        cfg.verify()?;
+        Ok(cfg)
+    }
+
+    fn load_from_file() -> Result<Self> {
+        // TODO: from where do we actually load the config?
```

Comment:
I think there should be a CLI flag like `--config <PATH>` and `./ts-rs.toml` or maybe `./ts-rs/config.toml` should be a default location if the flag isn't provided